src/platform/Darwin/DnssdImpl.h

```diff
@@ -102,6 +104,17 @@ struct RegisterContext : public GenericContext
     bool matches(const char * sType) { return mType.compare(sType) == 0; }
 };
 
+struct RegisterRecordContext : public GenericContext
+{
+    DnssdPublishCallback callback;
```

Comment:
Can we add a comment that this callback is *only* used if there is a failure when the record is registered ?

Reply:
done